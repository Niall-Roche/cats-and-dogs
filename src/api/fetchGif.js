import axios from 'axios'

export default async function (id) {
  try {
    const resp = await axios
      .get(`https://api.giphy.com/v1/gifs/${id}`, {
        params: {
          api_key: process.env.REACT_APP_GIPHY_API_KEY,
          gif_id: id,
        },
      })

    if (!!resp?.data) {
      return resp?.data
    }
  }
  catch (e) {
    return Promise.reject(e)
  }
}