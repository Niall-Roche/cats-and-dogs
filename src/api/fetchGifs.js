import axios from 'axios'

export default async function (q = 'cats') {
  try {
    const resp = await axios
      .get('https://api.giphy.com/v1/gifs/search', {
        params: {
          api_key: process.env.REACT_APP_GIPHY_API_KEY,
          q,
          limit: 25,
        },
      })

    if (!!resp?.data) {
      return resp?.data
    }
  }
  catch (e) {
    return Promise.reject(e)
  }
}