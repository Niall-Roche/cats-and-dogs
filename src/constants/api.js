export const LIMIT = 25
export const MAX_OFFSET = 4999
export const MAX_TOTAL = MAX_OFFSET + LIMIT
export const MAX_PAGES = Math.ceil(MAX_TOTAL / LIMIT)