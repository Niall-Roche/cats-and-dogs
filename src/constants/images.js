import CAT_EMOJI from '../assets/images/cat.png'
import DOG_EMOJI from '../assets/images/dog.png'
import GIPHY_ICON from '../assets/images/giphy-icon.png'
import INSTAGRAM_ICON from '../assets/images/instagram-icon.png'

export {
  CAT_EMOJI,
  DOG_EMOJI,
  GIPHY_ICON,
  INSTAGRAM_ICON,
}
