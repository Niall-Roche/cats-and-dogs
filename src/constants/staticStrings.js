export const CATS_QUERY = 'cats'
export const DOGS_QUERY = 'dogs'