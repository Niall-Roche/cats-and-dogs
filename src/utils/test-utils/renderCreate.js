import {ThemeProvider} from '@tanstack/react-query-devtools/build/lib/theme'
import TestRenderer from 'react-test-renderer'
import theme from '../../app-theme.json'
import {CatsAndDogsProvider} from '../../hooks/useCatsAndDogs'
import {PaginationProvider} from '../../hooks/usePagination'

export default comp => TestRenderer.create(
  <CatsAndDogsProvider>
    <PaginationProvider>
      <ThemeProvider theme={theme}>
        {comp}
      </ThemeProvider>
    </PaginationProvider>
  </CatsAndDogsProvider>
)