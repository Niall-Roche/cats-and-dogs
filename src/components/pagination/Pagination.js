import React from 'react'
import {func, number, string} from 'prop-types'
import {LIMIT, MAX_PAGES} from '../../constants/api'
import StyledPagination from './StyledPagination'

const propTypes = {
  className: string,
  limit: number,
  onPageChange: func,
  total: number,
}

const defaultProps = {
  className: '',
  limit: LIMIT,
  onPageChange: () => console.log('on page change'),
  total: 0,
}

const Pagination = ({className, limit, onPageChange, total, ...rest}) => {
  const pageCount = Math.ceil(total / limit)
  const totalPageCount = pageCount > MAX_PAGES ? MAX_PAGES : pageCount

  return (
    totalPageCount > 1
      ? (
        <div className={className}>
          <StyledPagination
            onPageChange={({selected}) => {
              const pageNumber = selected + 1
              onPageChange(pageNumber)
            }}
            pageCount={totalPageCount}
            {...rest}
          />
        </div>
      )
      : null
  )
}

Pagination.propTypes = propTypes
Pagination.defaultProps = defaultProps

export default Pagination

export const calculateOffset = (pageNumber, limit = LIMIT) => {
  if (pageNumber === 1) return 0

  return (pageNumber - 1) * limit
}
