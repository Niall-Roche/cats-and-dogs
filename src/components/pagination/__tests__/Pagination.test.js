import React from 'react'
import Pagination from '../Pagination'
import 'jest-styled-components'
import renderCreate from '../../../utils/test-utils/renderCreate'
import {LIMIT} from '../../../constants/api'

describe('Pagination', () => {
  it('should match snapshot', () => {
    const comp = renderCreate(<Pagination />)
    const tree = comp.toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('should match snapshot with required props', () => {
    const comp = renderCreate(
      <Pagination
        className='d-flex'
        limit={LIMIT}
        onPageChange={() => {}}
        total={0}
      />
    )
    const tree = comp.toJSON()

    expect(tree).toMatchSnapshot()
  })
})