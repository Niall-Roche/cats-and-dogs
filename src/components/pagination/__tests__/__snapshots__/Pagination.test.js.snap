// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination should match snapshot 1`] = `null`;

exports[`Pagination should match snapshot with required props 1`] = `null`;
